providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java

```diff
@@ -0,0 +1,203 @@
+package dev.openfeature.contrib.providers.flagd.resolver.process;
+
+import dev.openfeature.contrib.providers.flagd.FlagdOptions;
+import dev.openfeature.contrib.providers.flagd.resolver.Resolver;
+import dev.openfeature.contrib.providers.flagd.resolver.process.model.FeatureFlag;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.FlagStore;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.Storage;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.StorageState;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.grpc.GrpcStreamConnector;
+import dev.openfeature.sdk.ErrorCode;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Reason;
+import dev.openfeature.sdk.Value;
+import io.github.jamsesso.jsonlogic.JsonLogic;
+import io.github.jamsesso.jsonlogic.JsonLogicException;
+import lombok.extern.java.Log;
+
+import java.util.function.Consumer;
+import java.util.logging.Level;
+
+import static dev.openfeature.contrib.providers.flagd.resolver.process.model.FeatureFlag.EMPTY_TARGETING_STRING;
+
+/**
+ * flagd in-process resolver. Resolves feature flags in-process. Flags are retrieved from {@link Storage}, where the
+ * {@link Storage} maintain flag configurations obtained from known source.
+ */
+@Log
+public class InProcessResolver implements Resolver {
+    private final Storage flagStore;
+    private final Consumer<ProviderState> stateConsumer;
+    private final JsonLogic jsonLogicHandler;
+
+    /**
+     * Initialize an in-process resolver.
+     */
+    public InProcessResolver(FlagdOptions options, Consumer<ProviderState> stateConsumer) {
+        // currently we support gRPC connector
+        this.flagStore = new FlagStore(new GrpcStreamConnector(options));
+        this.stateConsumer = stateConsumer;
+        jsonLogicHandler = new JsonLogic();
+    }
+
+    /**
+     * Initialize in-process resolver.
+     */
+    public void init() throws Exception {
+        flagStore.init();
+        final Thread stateWatcher = new Thread(() -> {
+            try {
+                while (true) {
+                    final StorageState storageState = flagStore.getStateQueue().take();
+                    switch (storageState) {
+                        case OK:
+                            stateConsumer.accept(ProviderState.READY);
```

Comment:
If I'm understanding correctly, if we are already `READY`, this will emit a PROVIDER_CONFIGURATION_CHANGED event due to [this](https://github.com/open-feature/java-sdk-contrib/pull/412/files#diff-7ac82836e0450dd1f6581536670fba629b6fa21a8ccb4a5ffcdce54a53dd57e3R137-R143).

Reply:
Yup, this is correct. READY -> READY means there is a fresh configuration update.